docs/proposals/20190610-machine-states-preboot-bootstrapping.md

```diff
@@ -0,0 +1,438 @@
+      * [As an infrastructure provider consumer, I would like to have validation for the provider\-specific data I need to give the system to have it provision a machine\.](#as-an-infrastructure-provider-consumer-i-would-like-to-have-validation-for-the-provider-specific-data-i-need-to-give-the-system-to-have-it-provision-a-machine)
+      * [As an infrastructure  provider author, I would like to build a controller to manage provisioning machines using tools of my own choosing\.](#as-an-infrastructure--provider-author-i-would-like-to-build-a-controller-to-manage-provisioning-machines-using-tools-of-my-own-choosing)
+      * [As an infrastructure provider author, I would like to build a controller to manage provisioning machines without being restricted to a CRUD API\.](#as-an-infrastructure-provider-author-i-would-like-to-build-a-controller-to-manage-provisioning-machines-without-being-restricted-to-a-crud-api)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Machine Controller Role](#machine-controller-role)
+      * [Machine Controller dynamic watchers](#machine-controller-dynamic-watchers)
+      * [Object References, Templates, MachineSets and MachineDeployments](#object-references-templates-machinesets-and-machinedeployments)
+      * [Controllers and the single responsibility approach](#controllers-and-the-single-responsibility-approach)
+      * [Remote references and accessing a workload cluster](#remote-references-and-accessing-a-workload-cluster)
+      * [The “Phase” field and its role](#the-phase-field-and-its-role)
+      * [Showing a status summary to users](#showing-a-status-summary-to-users)
+    * [Risks and Mitigations](#risks-and-mitigations)
+      * [State transitions are inflexible](#state-transitions-are-inflexible)
+      * [Machine Controller can access any machine or cluster in any namespace](#machine-controller-can-access-any-machine-or-cluster-in-any-namespace)
+      * [Certificates and tokens are exposed in plaintext](#certificates-and-tokens-are-exposed-in-plaintext)
+      * [Bootstrap data cannot be merged](#bootstrap-data-cannot-be-merged)
+      * [MachineClass is deprecated and will be revisited later](#machineclass-is-deprecated-and-will-be-revisited-later)
+  * [Design Details](#design-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+    * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    * [Version Skew Strategy](#version-skew-strategy)
+  * [Implementation History](#implementation-history)
+  * [Drawbacks](#drawbacks)
+  * [Alternatives](#alternatives)
+      * [Object References, Templates, MachineSets and MachineDeployments](#object-references-templates-machinesets-and-machinedeployments-1)
+
+## Glossary
+
+- **Cluster API**: Unless otherwise specified, this refers to the project as a whole.
```

Comment:
I don't know how you feel about this, but I made the terms in the glossary targets so we could link directly to them, like this:

```suggestion
- **[Cluster API](../glossary.md#cluster-api)**: Unless otherwise specified, this refers to the project as a whole.
```